### PR TITLE
Add requires_php support

### PR DESCRIPTION
**Pull request by renatho**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR is a dependency for another change in the _Woo helper API_, which checks the required PHP version for WooCommerce extensions and avoid updates if the user contains unsupported PHP version. It uses the WordPress feature introduced in https://core.trac.wordpress.org/ticket/44350 and https://core.trac.wordpress.org/ticket/40934.

### How to test the changes in this Pull Request:

1. Remove the transient `_woocommerce_helper_updates` and the site transient `update_plugins`.
2. Install any WooCommerce extension.
3. Make sure your [WooCommerce.com Subscriptions](https://docs.woocommerce.com/document/managing-woocommerce-com-subscriptions/) is connected.
4. Intercept through the `wp_safe_remote_request` the WooCommerce endpoint that checks the plugins to update (`https://woocommerce.com/wp-json/helper/1.0/update-check?token=[xxxxxxxxxxxx]`) and returns one of the items with `requires_php: '8.0'` (set the version to be bigger than your environment PHP version). Or you can just force it in the code with `$item['requires_php'] = 8.0` instead of the `if`.
5. Reload the screen _WP Admin -> Plugins -> Installed Plugins_.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

Add the ability to avoid unsupported updates based on the PHP version.

### Screenshot

Expected result in the screen:

Screen Shot 2020-10-30 at 18 28 55

### CODE REVIEW

**Review by haszari — COMMENTED** [review]
Code looks safe to me. Having this in place will be useful 👍 Note I haven't tested this. 

Possibly could also test by connecting your test woo site to a local WCCOM dev site and setting `requires_php` for the extension in the WCCOM side.

I'll leave this to @woocommerce/proton to review & coordinate releasing this with the other components (WCCOM, WP core).

**Review by bgrgicak — DISMISSED** [review]
Works as expected! I tested it together with [wccom changes(https://github.com/Automattic/woocommerce.com/pull/5742) and the update worked.

**Comment by renatho** [comment]
Thank you for the review @haszari and @bgrgicak !
Just to let you know I don't have the permissions to merge this branch, so I wait for someone that can merge this. 😉

File: includes/admin/helper/class-wc-helper-updater.php
Comment (rodrigoprimo) [inline includes/admin/helper/class-wc-helper-updater.php:60]: Is there any chance that `$data['requires_php']` is not set and thus we should check if it is set to avoid a undefined index notice?

**Review by rodrigoprimo — COMMENTED** [review]
@renatho, the code changes look good to me. I just have one quick question before merging this that I left in a inline comment.

Noting that I didn't test this PR since others already did.

File: includes/admin/helper/class-wc-helper-updater.php
Comment (renatho) [reply includes/admin/helper/class-wc-helper-updater.php]: Hey @rodrigoprimo! Thank you for your review!

Good catch! It's really possible! I updated it here: https://github.com/woocommerce/woocommerce/pull/28151/commits/142f810206edff672331883a60e83592bc0d79f4

**Review by rodrigoprimo — APPROVED** [review]
Looks good to me. Thanks for the change, @renatho. Leaving it unmerged as you requested a review from other people. Let me know when this is ready to be merged and I will do it.

Just noting that the Travis failure is not related to this PR. It is a problem in Composer 2 and Jetpack Autoloader that we are looking into fixing as soon as possible.

**Review by bgrgicak — APPROVED** [review]

**Comment by renatho** [comment]
@rodrigoprimo, considering the last change was a tiny tweak, and there are already 2 approvals, I think we could merge. =)

**Pull request merged**
